### PR TITLE
Allow for building without QtKeychain

### PR DESCRIPTION
**Pull request by Mm2PL**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description

This doesn't support QMake, because I don't really see the value in putting in work in fighting another build system, especially that it's mostly used on Windows.

### CODE REVIEW

**Review by pajlada — APPROVED** [review]
:+1: I've sent the patch over to vhns for testing to ensure the changes I made don't interfere (so let's not merge it in yet)

**Comment by vhns** [comment]
Worked just fine. I can change the Gentoo package to no longer use the patch once merged uptream.

**Review by pajlada — APPROVED** [review]

**Pull request merged**
